src/throttler/throttler.ts

```diff
@@ -138,16 +163,17 @@ export abstract class Throttler<T> {
       this.success++;
       this.complete++;
       this.active--;
-      delete this.tasks[cursorIndex];
-      delete this.retryCounts[cursorIndex];
+      if (taskData.wait) {
+        taskData.wait.resolve(result);
+      }
+      delete this.taskDatas[cursorIndex];
       this.process();
     } catch (err) {
       if (this.retries > 0) {
-        this.retryCounts[cursorIndex] = this.retryCounts[cursorIndex] || 0;
-        if (this.retryCounts[cursorIndex] < this.retries) {
-          this.retryCounts[cursorIndex]++;
+        if (taskData.retryCount < this.retries) {
+          taskData.retryCount++;
           this.retried++;
-          await _backoff(this.retryCounts[cursorIndex], this.backoff);
+          await _backoff(taskData.retryCount, this.backoff);
```

Comment:
I'm sorry I didn't catch this before:

Don't prepend private function names/values with `_`. They're already marked private, so the compiler will take care of that for us.